src/hotspot/share/gc/g1/g1SATBMarkQueueSet.cpp

```diff
@@ -97,24 +97,9 @@ static inline bool requires_marking(const void* entry, G1CollectedHeap* g1h) {
   return true;
 }
 
-static inline bool discard_entry(const void* entry, G1CollectedHeap* g1h) {
-  return !requires_marking(entry, g1h) || g1h->is_marked_next(cast_to_oop(entry));
-}
-
-// Workaround for not yet having std::bind.
-class G1SATBMarkQueueFilterFn {
-  G1CollectedHeap* _g1h;
-
-public:
-  G1SATBMarkQueueFilterFn() : _g1h(G1CollectedHeap::heap()) {}
-
-  // Return true if entry should be filtered out (removed), false if
-  // it should be retained.
-  bool operator()(const void* entry) const {
-    return discard_entry(entry, _g1h);
-  }
-};
-
 void G1SATBMarkQueueSet::filter(SATBMarkQueue& queue) {
-  apply_filter(G1SATBMarkQueueFilterFn(), queue);
+  auto requires_discard = [&] (const void* entry) {
+    return !requires_marking(entry, _g1h) || _g1h->is_marked_next(cast_to_oop(entry));
+  };
+  apply_filter(requires_discard, queue);
```

Comment:
This on the other hand, is a nice usage and saves a lot of boilerplate, and it's not like operator() was that easy to read in the first place.